### ISSUE TITLE
Undefined is not an object (evaluating 'RNGoogle.SignIn.Button_SIZE_ICON)

### ISSUE THREAD

**Issue by Jarrio** [haxe-react/haxe-react-native-google-signin#1]
Wanted to mess around with this but was wondering if it's fully been ported, I attempted to apply a function to a button which called `GoogleSignin.signIn()` which returns the error in the title. I can't seem to find any reference to being able to use the proper google button as well
```  
<GoogleSigninButton
    style={{width: 48, height: 48}}
    size={GoogleSigninButton.Size.Icon}
    color={GoogleSigninButton.Color.Dark}
    onPress={this._signIn.bind(this)}
/>
```

which I assume is the issue here?

**Comment by kevinresol** [comment]
The extern for the button is missing. I think adding the following code should work

```haxe
@:jsRequire('react-native-google-signin', 'GoogleSigninButton')
extern class GoogleSigninButton extends react.ReactComponent {}
```

**Comment by Jarrio** [comment]
Just tried it and following the example on the `react-native-google-signin` page it returns the following error (within haxe compiler) 

```
src/scene/Elements.hx:69: lines 69-97 : Class<react.native.google_signin.GoogleSigninButton> has no field Size
src/scene/Elements.hx:69: lines 69-97 : Class<react.native.google_signin.GoogleSigninButton> has no field Color
```

**Comment by kevinresol** [comment]
would need to define the fields too. For starter, you can use `Dynamic`

```haxe
@:jsRequire('react-native-google-signin', 'GoogleSigninButton')
extern class GoogleSigninButton extends react.ReactComponent {
  var Size:Dynamic;
  var Color:Dynamic;
}
```

**Comment by Jarrio** [comment]
I added the properties, although I had to make them static before it was recognised. I still receive the same error (`undefined is not an object`) within the app. It refers to line 40 of the `GoogleSignin.android.js` file in the react-native side of things. Which is the following: `Icon: RNGoogleSignin.BUTTON_SIZE_ICON`

Where the whole segment is:

```
GoogleSigninButton.Size = {
  Icon: RNGoogleSignin.BUTTON_SIZE_ICON,
  Standard: RNGoogleSignin.BUTTON_SIZE_STANDARD,
  Wide: RNGoogleSignin.BUTTON_SIZE_WIDE
};

GoogleSigninButton.Color = {
  Auto: RNGoogleSignin.BUTTON_COLOR_AUTO,
  Light: RNGoogleSignin.BUTTON_COLOR_LIGHT,
  Dark: RNGoogleSignin.BUTTON_COLOR_DARK
};
```

Where both things are outside of classes if that makes any difference

**Comment by kevinresol** [comment]
That looks like a missing native library. Did you install & link the native things properly?

**Comment by Jarrio** [comment]
I was using the already running server and didn't re-run `react-native run-android` after doing that I got errors saying my project wasn't working because of invalid file path lengths and that was preventing the google-signin react-native plugin to get added to the application.

I moved the project to the root directory of my drive and that fixed the issue (Including the fixes to the haxe lib itself)! Thanks!
